tests/spaces/test_box.py

```diff
@@ -57,9 +58,11 @@ def test_box_shape_inference(box, expected_shape):
 )
 def test_box_values(value, valid):
     if valid:
-        with pytest.warns(None) as warnings:
+        with warnings.catch_warnings(record=True) as caught_warnings:
             Box(low=value, high=value)
-        assert len(warnings.list) == 0, tuple(warning.message for warning in warnings)
+        assert len(caught_warnings) == 0, tuple(
+            warning.message for warning in caught_warnings
+        )
```

Comment:
How about replacing this with
```python
with warnings.catch_warnings():
    warnings.simplefilter("error")
    Box(low=value, high=value)
```
Cf https://docs.pytest.org/en/7.0.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests

However, I believe that this should only be done in cases in which we do *not* also use `pytest.raises`, because otherwise `pytest.raises` might be satisfied with a warning that was promoted to an exception.

Reply:
As we are adding this to a limited number of cases then I dont think this should matter